src/adapter/graphql/src/mutations/account_mut.rs

```diff
@@ -135,9 +109,28 @@ impl AccountMut {
                 ModifyPasswordSuccess::default(),
             )),
             Err(ModifyPasswordError::AccountNotFound(_)) => Ok(
-                ModifyPasswordResult::AccountNotFound(ModifyPasswordAccountNotFound::default()),
+                ModifyPasswordResult::AccountNotFound(AccountNotFound::default()),
             ),
-            Err(ModifyPasswordError::Internal(e)) => Err(e.into()),
+            Err(e @ ModifyPasswordError::Internal(_)) => Err(e.int_err().into()),
+        }
+    }
+
+    /// Deleting an account by name
+    async fn delete_account_by_name(
+        &self,
+        ctx: &Context<'_>,
+        account_name: AccountName<'_>,
+    ) -> Result<DeleteAccountResult> {
+        ensure_account_can_provision_accounts(ctx, &self.account.id).await?;
```

Comment:
Unlikely it's the same permission?

Reply:
You're right, it's not the same.

Currently, the following: you can delete yourself or be an admin to delete a user